frontend/deb/template_rules.go

```diff
@@ -67,6 +67,88 @@ func (w *rulesWrapper) OverridePerms() fmt.Stringer {
 	artifacts := w.GetArtifacts(w.target)
 
 	var fixPerms bool
+	// also need to check for fields that are of type map[string]dalec.ArtifactConfig
+	// because they can also have a Mode field.
+
+	for _, cfg := range artifacts.Binaries {
+		if cfg.Mode != 0 {
```

Comment:
Is there a way to D.R.Y. this up?

Something like create `artifactTypes` that's a slice of slices of type dalec.ArtifactConfig(?), iterate through those types then check if the mode field is there for each one (which itself could be it's own function).

Would remove a lot of this redundancy. Wdyt?